packages/core/src/agent/didcomm/types.ts

```diff
@@ -0,0 +1,81 @@
+import type { Key } from '../../crypto'
+
+export enum MessageType {
+  Plain = 'plain',
+  Signed = 'signed',
+  Encrypted = 'encrypted',
+}
+
+export type ReceivedMessage = EncryptedMessage | SignedMessage | PlaintextMessage
+
+export type PlaintextMessage = PlaintextDIDCommV1Message | PlaintextDIDCommV2Message
+
+export interface PlaintextDIDCommV1Message {
+  '@type': string
+  '@id': string
+  [key: string]: unknown
+}
+
+export interface PlaintextDIDCommV2Message {
+  type: string
+  id: string
+  [key: string]: unknown
+}
+
+export type EncryptedMessageRecipientHeader = {
+  kid: string
+}
+
+export type EncryptedMessageRecipient = {
+  encrypted_key: string
+  header: EncryptedMessageRecipientHeader
+}
+
+export type EncryptedMessage = {
+  protected: string
+  iv: string
+  ciphertext: string
+  tag: string
+  recipients: EncryptedMessageRecipient[]
+}
+
+export type SignatureHeader = {
+  kid: string
+}
+
+export type Signature = {
+  header: SignatureHeader
+  protected: string
+  signature: string
+}
+
+export type SignedMessage = {
+  signatures: Array<Signature>
+  payload: string
+}
+
+export type ProtectedMessage = {
+  typ: string
+  alg: string
+}
+
+export enum DIDCommMessageVersion {
+  V1 = 'DIDCommV1',
+  V2 = 'DIDCommV2',
+}
+
+export enum DidCommV1Types {
+  JwmV1 = 'JWM/1.0',
+}
+
+export enum DidCommV1Algorithms {
+  Authcrypt = 'Authcrypt',
+  Anoncrypt = 'Anoncrypt',
+}
+
+export interface DecryptedMessageContext {
+  plaintextMessage: PlaintextMessage
+  senderKey?: Key
+  recipientKey?: Key
+  version?: DIDCommMessageVersion
```

Comment:
Is version determine by the message content version or the envelope format version? 

```suggestion
  didCommVersion?: DIDCommMessageVersion
```


Message content could be v1 and envelope could be v2

<img width="881" alt="image" src="https://user-images.githubusercontent.com/23165168/202182862-5a32a4b1-5a97-4309-a180-600113bb6252.png">